### PR TITLE
Implements during ReQL command

### PR DESCRIPTION
**Pull request by zedtux**
This PR implements [the `during` ReQL command](https://rethinkdb.com/api/ruby/during/) which allow to filter documents having the given filed between the 2 given times.

Here is an example of its usage:

```ruby
User.where(:created_at.during(1.month.ago, Time.current)).to_a
```

It returns all the created `User`s in the last month period of time.

It replaces using the combinaison of `gt` and `lt`.

### CODE REVIEW

**Comment by zedtux** [comment]
There's an issue with type casting so please wait before to more this one. Sorry about that.

File: spec/integration/criteria/where_spec.rb [hunk 1]
```diff
@@ -28,7 +28,7 @@
   context 'when using bad arguments' do
     it 'raises' do
       expect { SimpleDocument.where("xxx") }.to raise_error(/Invalid/)
-      expect { SimpleDocument.where(:xxx.eq(1,2)) }.to raise_error(/Invalid/)
```
Comment (nviennot) [inline spec/integration/criteria/where_spec.rb:31]: eq(1,2) should still raise an error

Comment (zedtux) [reply spec/integration/criteria/where_spec.rb:31]: Nice catch! Of course. I'll change that, thank you!

**Comment by zedtux** [comment]
@nviennot I fixed your review and added the TimeWithZone support for `during`.

When you'll have accepted it, I can submit a PR to update the doc if you'd like to 😉

**Pull request merged**